Source/DOM classes/SVG-DOM/SVGHelperUtilities.m

```diff
@@ -318,20 +318,37 @@ +(CALayer *) newCALayerForPathBasedSVGElement:(SVGElement<SVGTransformable>*) sv
 	
 	[self configureCALayer:_shapeLayer usingElement:svgElement];
 	
+	NSString* actualStroke = [svgElement cascadedValueForStylableProperty:@"stroke"];
+	if (!actualStroke)
+		actualStroke = @"none";
+	NSString* actualStrokeWidth = [svgElement cascadedValueForStylableProperty:@"stroke-width"];
+	
+	CGFloat strokeWidth = 1.0;
+	
+	if (actualStrokeWidth)
+	{
+		SVGRect r = ((SVGSVGElement*) svgElement.viewportElement).viewport;
+		
+		strokeWidth = [[SVGLength svgLengthFromNSString:actualStrokeWidth]
+					   pixelsValueWithDimension: hypot(r.width, r.height)];
+	}
+	
 	/** transform our LOCAL path into ABSOLUTE space */
 	CGAffineTransform transformAbsolute = [self transformAbsoluteIncludingViewportForTransformableOrViewportEstablishingElement:svgElement];
-	CGMutablePathRef pathToPlaceInLayer = CGPathCreateMutable();
-	CGPathAddPath( pathToPlaceInLayer, &transformAbsolute, pathRelative);
+
+	// calculate the rendered dimensions of the path
+	CGRect r = CGRectInset(CGPathGetBoundingBox(pathRelative), -strokeWidth/2., -strokeWidth/2.);
+	CGRect transformedPathBB = CGRectApplyAffineTransform(r, transformAbsolute);
+	
```

Comment:
The bound box is outset by 1/2 the line width before the absolute transformation is applied.